gridstatus/ieso.py

```diff
@@ -844,3 +911,887 @@ def _request(self, url, verbose):
+                    )
+
+            if "reserves" in section_data:
+                logger.debug("Processing Generation Reserve Holdback Data...")
+                reserves_config = section_data["reserves"]
+
+                current_data = document_body
+                for path_part in reserves_config["path"]:
+                    current_data = current_data[path_part]
+
+                for section_name, section_config in reserves_config["sections"].items():
+                    self._extract_hourly_values(
+                        data=current_data,
+                        path=[section_config["container"], section_config["item_key"]],
+                        column_name=section_name,
+                        value_key="EnergyMW",
+                        report_data=report_data,
+                    )
+
+            if "ontario_demand" in section_data:
+                logger.debug("Processing Ontario Demand Data...")
+                ontario_demand_config = section_data["ontario_demand"]
+
+                current_data = document_body
+                for path_part in ontario_demand_config["path"]:
+                    current_data = current_data[path_part]
+
+                for section_name, section_config in ontario_demand_config[
+                    "sections"
+                ].items():
+                    if "sections" in section_config:
+                        continue
+
+                    section_data = current_data
+                    self._extract_hourly_values(
+                        data=section_data,
+                        path=section_config["path"],
+                        column_name=section_name,
+                        value_key=section_config["value_key"],
+                        report_data=report_data,
+                    )
+
+                for ontario_demand_btm in [
+                    "Dispatchable Load",
+                    "Hourly Demand Response",
+                ]:
+                    btm_config = ontario_demand_config["sections"][ontario_demand_btm]
+                    btm_data = current_data[ontario_demand_btm.replace(" ", "")]
+
+                    for section_name, section_config in btm_config["sections"].items():
+                        self._extract_hourly_values(
+                            data=btm_data,
+                            path=[
+                                section_config["container"],
+                                section_config["item_key"],
+                            ],
+                            column_name=section_name,
+                            value_key=section_config["value_key"],
+                            report_data=report_data,
+                        )
+
+        # NOTE(kladar): This is the first place where pandas is truly invoked, leaving it open for more modern
+        # dataframe libraries to be swapped in in the future
+        df = pd.DataFrame(report_data)
+        df["Interval Start"] = delivery_date + pd.to_timedelta(
```

Comment:
I think we want to add the timezone to "Interval Start", "Interval End", and "Publish Time".

Reply:
sounds good. 

I've added that when they are parsed from the document and moved that parsing to the rest of the date handling here as well.